src/errors/validation_exception.rs

```diff
@@ -194,8 +194,16 @@ impl ValidationError {
 static URL_ENV_VAR: GILOnceCell<bool> = GILOnceCell::new();
 
 fn _get_include_url_env() -> bool {
-    match std::env::var("PYDANTIC_ERRORS_OMIT_URL") {
-        Ok(val) => val.is_empty(),
+    if std::env::var_os("PYDANTIC_ERRORS_OMIT_URL").is_some() {
+        // This is ugly (and is interposed weirdly into a printed traceback), but
+        // there probably isn't a much better way to surface a deprecation warning
+        // from so deep in the code. Hopefully there hadn't been too many users
+        // using the undocumented environment variable who will need to suffer
+        // the ugliness, and when they do, they'll act accordingly.
+        eprintln!("_pydantic_core: PYDANTIC_ERRORS_OMIT_URL is deprecated, use PYDANTIC_ERRORS_INCLUDE_URL instead");
```

Comment:
Potentially can use `PyErr::warn` here to be a bit less ugly?

I think if this environment variable is set, we should use it and invert its value, as a sort of backwards compatibility. That will probably fix the integration test failure.

Reply:
Done deal. Needed a bit of rejiggling to get the `py` handle into the `get_or_init` closure.